src/main/java/com/synopsys/integration/jira/common/server/service/IssueService.java

```diff
@@ -0,0 +1,188 @@
+import com.synopsys.integration.jira.common.model.components.ProjectComponent;
+import com.synopsys.integration.jira.common.model.components.StatusDetailsComponent;
+import com.synopsys.integration.jira.common.model.request.IssueCommentRequestModel;
+import com.synopsys.integration.jira.common.model.request.IssueRequestModel;
+import com.synopsys.integration.jira.common.model.request.JiraCloudRequestFactory;
+import com.synopsys.integration.jira.common.model.response.IssueResponseModel;
+import com.synopsys.integration.jira.common.model.response.IssueTypeResponseModel;
+import com.synopsys.integration.jira.common.model.response.TransitionsResponseModel;
+import com.synopsys.integration.jira.common.model.response.UserDetailsResponseModel;
+import com.synopsys.integration.jira.common.rest.service.IssueTypeService;
+import com.synopsys.integration.jira.common.rest.service.JiraService;
+import com.synopsys.integration.jira.common.server.builder.IssueRequestModelFieldsBuilder;
+import com.synopsys.integration.jira.common.server.model.IssueCreationRequestModel;
+import com.synopsys.integration.rest.request.Request;
+import com.synopsys.integration.rest.request.Response;
+import com.synopsys.integration.rest.service.IntJsonTransformer;
+
+public class IssueService {
+    public static final String API_PATH = "/rest/api/2/issue";
+    public static final String API_PATH_TRANSITIONS_SUFFIX = "transitions";
+    public static final String API_PATH_COMMENTS_SUFFIX = "comment";
+
+    private final IntJsonTransformer intJsonTransformer;
+    private final JiraService jiraService;
+    private final UserSearchService userSearchService;
+    private final ProjectService projectService;
+    private final IssueTypeService issueTypeService;
+
+    public IssueService(IntJsonTransformer intJsonTransformer, JiraService jiraService, UserSearchService userSearchService, ProjectService projectService, IssueTypeService issueTypeService) {
+        this.intJsonTransformer = intJsonTransformer;
+        this.jiraService = jiraService;
+        this.userSearchService = userSearchService;
+        this.projectService = projectService;
+        this.issueTypeService = issueTypeService;
+    }
+
+    public IssueResponseModel createIssue(IssueCreationRequestModel requestModel) throws IntegrationException {
+        String issueTypeName = requestModel.getIssueTypeName();
+        String projectName = requestModel.getProjectName();
+        String reporter = requestModel.getReporterUsername();
+
+        IssueTypeResponseModel foundIssueType = issueTypeService.getAllIssueTypes().stream()
+                                                    .filter(issueType -> issueType.getName().equalsIgnoreCase(issueTypeName))
+                                                    .findFirst()
+                                                    .orElseThrow(() -> new IllegalStateException(String.format("Issue type not found; issue type %s", issueTypeName)));
```

Comment:
Should we be throwing RuntimeExceptions here? Or should we throw something that extends IntegrationException?